packages/flutter_tools/lib/src/windows/visual_studio_solution_utils.dart

```diff
@@ -251,7 +252,17 @@ EndProject\r
   /// in [plugins].
   Map<String, String> _getWindowsPluginNamesByGuid(List<Plugin> plugins) {
     final Map<String, String> currentPluginInfo = <String, String>{};
-    for (final Plugin plugin in plugins) {
+    // Plugins without a 'pluginClass' definition shouldn't generate a GUID. The plugin might
+    // be a pure Dart implementation.
+    final List<Plugin> nativeWindowsPlugins = plugins.where((final Plugin element) {
+      final WindowsPlugin windowsPlugin = element.platforms[WindowsPlugin.kConfigKey] as WindowsPlugin;
+      if (windowsPlugin == null) {
+        return false;
+      }
+      return windowsPlugin.pluginClass != null && windowsPlugin.pluginClass.isNotEmpty;
```

Comment:
Why the `isNotEmpty` clauses? If someone has a plugin with an empty pluginClass entry, they have screwed up; we shouldn't just silently drop it.